sdk/keyvault/keyvault-keys/src/cryptography/rsaCryptographyProvider.ts

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { RSA_PKCS1_OAEP_PADDING, RSA_PKCS1_PADDING } from "constants";
+import { JsonWebKey, KeyOperation, KnownKeyOperations } from "../keysModels";
+import { publicEncrypt } from "crypto";
+import { createVerify } from "./hash";
+import {
+  EncryptOptions,
+  EncryptParameters,
+  EncryptResult,
+  KeyWrapAlgorithm,
+  VerifyOptions,
+  VerifyResult,
+  WrapKeyOptions,
+  WrapResult
+} from "..";
+import { convertJWKtoPEM } from "./conversions";
+import {
+  CryptographyProvider,
+  LocalCryptographyUnsupportedError,
+  LocalSupportedAlgorithmName
+} from "./models";
+import {
+  DecryptOptions,
+  DecryptParameters,
+  DecryptResult,
+  SignatureAlgorithm,
+  SignOptions,
+  SignResult,
+  UnwrapKeyOptions,
+  UnwrapResult
+} from "../cryptographyClientModels";
+
+/**
+ * An RSA cryptography provider supporting RSA algorithms.
+ */
+export class RsaCryptographyProvider implements CryptographyProvider {
+  constructor(key: JsonWebKey) {
+    this.key = key;
+  }
+
+  supportsAlgorithm(algorithm: LocalSupportedAlgorithmName): boolean {
+    return this.applicableAlgorithms.includes(algorithm);
+  }
+
+  supportsOperation(operation: KeyOperation): boolean {
+    return this.applicableOperations.includes(operation);
+  }
+
+  encrypt(encryptParameters: EncryptParameters, _options: EncryptOptions): Promise<EncryptResult> {
+    this.ensureValid();
+    const keyPEM = convertJWKtoPEM(this.key);
+
+    const padding =
+      encryptParameters.algorithm === "RSA1_5" ? RSA_PKCS1_PADDING : RSA_PKCS1_OAEP_PADDING;
+
+    return Promise.resolve({
+      algorithm: encryptParameters.algorithm,
+      keyID: this.key.kid,
+      result: publicEncrypt(
+        { key: keyPEM, padding: padding },
+        Buffer.from(encryptParameters.plaintext)
+      )
+    });
+  }
+
+  decrypt(_decryptParameters: DecryptParameters, _options: DecryptOptions): Promise<DecryptResult> {
+    throw new LocalCryptographyUnsupportedError(
+      "Decrypting using a local JsonWebKey is not supported."
+    );
+  }
+
+  wrapKey(
+    algorithm: KeyWrapAlgorithm,
+    keyToWrap: Uint8Array,
+    _options: WrapKeyOptions
```

Comment:
do you need to add this parameter if it's not used here? I think it may compile without it (also applies for all of the other unused parameters).

Reply:
Quick update: I originally removed the unused parameters and marked this resolved but I since reverted that change so I should reply to the comment. I tried both approaches and I found having different parameter lists for the interface vs the implementation confusing to follow, and in the future if we need to make changes here it's nice to see what parameters are getting passed in. The thing that tripped me up the most was that now these will have different method signatures:

```ts
const provider: CryptographyProvider = new RsaCryptographyProvider()
provider.wrapKey(algo, buffer, {})
const rsaProvider: RsaCryptographyProvider = new RsaCryptographyProvider()
provider.wrapKey(algo, buffer, {}) // compile error
```
And I feel like I should be able to substitute them...

But I did make options optional at the interface level fwiw and cleaned up anything that could be optional for all implementations.